### DISCUSSION TITLE
Way to retain the same order across sessions in RDM task but randomised across participants?

### DISCUSSION THREAD

**Discussion by hariPurohit** [jspsych/jsPsych#1572] [Q&A]
Hi!

I am trying to run a RDM (random-dot-motion) task using three levels of coherence (0.10, 0.15, 0.20) and two levels of coherent direction (left, right) (based on Ratcliff & McKoon, 2008). My experiment has two sessions for each participant, with the order of the blocks and trials randomised across participants but the order remains the same within participant, for both sessions. 

Is there a way for me to code and save the randomisation order for the two sessions in a way to allow randomising it across participants while retaining an order within participant sessions? Something like a 'seed' value or other function within jsPsych?

Any help or pointers to relevant discussions would be very much appreciated!

**Comment by jodeleeuw** [comment]
The default random number generator in JS, `Math.random()`, is not seedable. This is what jsPsych currently uses for all the randomization done by things like `randomize_order: true` or `jsPsych.randomization.shuffle()`.

(It might be worth using a seedable RNG in a future version of jsPsych to provide an option that will work for this kind of situation. I'll open an issue to track this - #1573.)

In the meantime, I think you'll need to either do the randomization outside of jsPsych and then load in the list of trials, or save parameters for each trial in the first session and then load those in for the second.
